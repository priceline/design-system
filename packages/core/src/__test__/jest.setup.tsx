import '@testing-library/jest-dom'
import 'jest-styled-components'

global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function _raf(cb) {
    return setTimeout(cb, 0)
  }
