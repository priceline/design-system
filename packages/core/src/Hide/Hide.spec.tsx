import React from 'react'
import { render } from '../__test__/testing-library'
import { Hide } from './Hide'

describe('Hide', () => {
  test('renders', () => {
    const { asFragment } = render(<Hide />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders with breakpoint props', () => {
    const { rerender, asFragment } = render(<Hide xs />)
    expect(asFragment()).toMatchSnapshot()

    rerender(<Hide sm />)
    expect(asFragment()).toMatchSnapshot()

    rerender(<Hide md />)
    expect(asFragment()).toMatchSnapshot()

    rerender(<Hide lg />)
    expect(asFragment()).toMatchSnapshot()

    rerender(<Hide xl />)
    expect(asFragment()).toMatchSnapshot()

    rerender(<Hide xxl />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders with print prop', () => {
    const { asFragment } = render(<Hide print />)
    expect(asFragment()).toMatchSnapshot()
  })
})
