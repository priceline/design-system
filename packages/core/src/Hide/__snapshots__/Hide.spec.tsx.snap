// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hide renders 1`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

<div
    class="c0"
  >
    <div
      class=""
    />
  </div>
</DocumentFragment>
`;

exports[`Hide renders with breakpoint props 1`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

@media screen and (max-width:31.99em) {
  .c1 {
    display: none;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
  </div>
</DocumentFragment>
`;

exports[`Hide renders with breakpoint props 2`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

@media screen and (max-width:31.99em) {

}

@media screen and (min-width:32em) and (max-width:39.99em) {
  .c1 {
    display: none;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
  </div>
</DocumentFragment>
`;

exports[`Hide renders with breakpoint props 3`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

@media screen and (max-width:31.99em) {

}

@media screen and (min-width:32em) and (max-width:39.99em) {

}

@media screen and (min-width:40em) and (max-width:47.99em) {
  .c1 {
    display: none;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
  </div>
</DocumentFragment>
`;

exports[`Hide renders with breakpoint props 4`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

@media screen and (max-width:31.99em) {

}

@media screen and (min-width:32em) and (max-width:39.99em) {

}

@media screen and (min-width:40em) and (max-width:47.99em) {

}

@media screen and (min-width:48em) and (max-width:63.99em) {
  .c1 {
    display: none;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
  </div>
</DocumentFragment>
`;

exports[`Hide renders with breakpoint props 5`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

@media screen and (max-width:31.99em) {

}

@media screen and (min-width:32em) and (max-width:39.99em) {

}

@media screen and (min-width:40em) and (max-width:47.99em) {

}

@media screen and (min-width:48em) and (max-width:63.99em) {

}

@media screen and (min-width:64em) and (max-width:79.99em) {
  .c1 {
    display: none;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
  </div>
</DocumentFragment>
`;

exports[`Hide renders with breakpoint props 6`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

@media screen and (max-width:31.99em) {

}

@media screen and (min-width:32em) and (max-width:39.99em) {

}

@media screen and (min-width:40em) and (max-width:47.99em) {

}

@media screen and (min-width:48em) and (max-width:63.99em) {

}

@media screen and (min-width:64em) and (max-width:79.99em) {

}

@media screen and (min-width:80em) {
  .c1 {
    display: none;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
  </div>
</DocumentFragment>
`;

exports[`Hide renders with print prop 1`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

@media print {
  .c1 {
    display: none;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
  </div>
</DocumentFragment>
`;
